orc8r/cloud/go/services/metricsd/prometheus/prometheus-cache/cache/cache.go

```diff
@@ -147,18 +147,45 @@ func (c *MetricCache) clearMetrics() {
 	c.metricFamiliesByName = make(map[string]*familyAndMetrics)
 }
 
-func (c *MetricCache) exposeMetrics(metricFamiliesByName map[string]*familyAndMetrics) string {
-	respStr := strings.Builder{}
+func (c *MetricCache) exposeMetrics(metricFamiliesByName map[string]*familyAndMetrics, workers int) string {
+	fams := make(chan *familyAndMetrics, workers)
+	results := make(chan string, workers)
+	respStr := make(chan string, 1)
+
+	for i := 0; i < workers; i++ {
+		go processFamilyWorker(fams, results)
+	}
+
+	go processFamilyStringsWorker(results, respStr, len(metricFamiliesByName))
+
 	for _, fam := range metricFamiliesByName {
+		fams <- fam
+	}
+	close(fams)
+
+	return <-respStr
+}
+
+func processFamilyWorker(fams <-chan *familyAndMetrics, results chan<- string) {
+	for fam := range fams {
 		pullFamily := fam.popSortedDatapoints()
 		familyStr, err := familyToString(pullFamily)
 		if err != nil {
 			glog.Errorf("metric %s dropped. error converting metric to string: %v", *pullFamily.Name, err)
 		} else {
-			respStr.WriteString(familyStr)
+			results <- familyStr
 		}
 	}
-	return respStr.String()
+}
+
+func processFamilyStringsWorker(results <-chan string, respStrChannel chan<- string, total int) {
+	respStr := strings.Builder{}
+
+	for i := 0; i < total; i++ {
```

Comment:
This function relies on the length of results being at least as long as `total`. In `processFamilyWorkers` there is a path where a string is not submitted to the channel, meaning that you can't assume `len(results) == len(metricFamilesByName)`.